src/com/google/cose/EncryptMessage.java

```diff
@@ -77,7 +72,7 @@ public Builder withCiphertext(byte[] ciphertext) {
     }
 
     public Builder withRecipients(List<Recipient> recipients) {
-      this.recipients.addAll(recipients);
+      this.recipients = recipients;
```

Comment:
copying the list over would be a reference. Even if we generate a new list the objects inside would still be referring to the same objects as such
Hence, I just overwrote the list instead of maintaining a new one

Reply:
Yeah, it would have to be a deep copy.

I'm not a java person, so it's not clear to me if builders normally make deep copies or just pass references along. As a C++ guy, the notion of a bunch of "built" objects all holding mutable references back to the same things freaks me out :)